### PR TITLE
PMM-9690 Release Notes

### PR DESCRIPTION
**Pull request by rasika-chivate**
Started Authoring the Release Notes for 2.27.0

### CODE REVIEW

File: docs/release-notes/2.27.0.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+
+## Improvements
+
+
+
+## Bugs Fixed
+
+[PMM-9502] (https://jira.percona.com/browse/PMM-9502) - Fixed multiple JavaScript Common Vulnerabilities and Exposures (CVE) for PMM AMI setup.
```
Comment (atymchuk) [inline docs/release-notes/2.27.0.md:43]:
```suggestion
[PMM-9502] (https://jira.percona.com/browse/PMM-9502) - Fixed multiple JavaScript Common Vulnerabilities and Exposures (CVE) for PMM AMI setup wizard (works on AWS only).
```

File: docs/release-notes/2.27.0.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    - Advisor checks have been extended with two new query types: getDiagnosticData and replSetGetStatus. 
+    - Documentation about developing checks 
+    - DBaaS - Simplified experience in registering EKS cluster with kubeconfig generated by eksctl.
+
+* Components upgrade
+    - Grafana: PMM 2.27.0 has now migrated to Grafana 8.3.5. This version of Grafana is loaded with a gamut of exciting features. For more information, see [What's new in Grafana v8.0](https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v8-0/).
+    - VictoriaMetrics: VictoriaMetrics has been upgraded to 1.72.0.
+
+
+## New Features
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:46]: we again missing https://jira.percona.com/browse/PMM-8800

File: docs/release-notes/2.27.0.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+- [PMM-9718](https://jira.percona.com/browse/PMM-9718): Federated connections to Percona Platform.
+
+!!! caution alert alert-warning "Important"
+    If you want to connect your PMM instance to the Percona Platform, make sure to specify the following variables for your Docker container: 
+    - `PERCONA_TEST_SAAS_HOST=check-dev.percona.com`
+    - `PERCONA_TEST_CHECKS_PUBLIC_KEY=RWTkF7Snv08FCboTne4djQfN5qbrLfAjb8SY3/wwEP+X5nUrkxCEvUDJ`
+This will allow you to connect your PMM 2.27 Preview version to Percona Platform Development environment for testing purposes.
+
+- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Additional abilities for Mongo MongoDB Advisor Checks.
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:56]:
```suggestion
- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Additional abilities for MongoDB Advisor Checks.
```

**Review by denisok — CHANGES_REQUESTED** [review]

File: docs/release-notes/2.27.0.md (hunk 2 above)
Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md:46]: Its there in Release Highlights:     - DBaaS - Simplified experience in registering EKS cluster with kubeconfig generated by eksctl.

File: docs/release-notes/2.27.0.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    - Social Login/Registration
+    - **Enhanced UX for connecting PMM server to Percona Portal**: PMM now requires access-token authorization for connecting PMM instances to Percona Platform.
+    - **More visibility for Percona customers**: Percona customers who connect their PMM instances to the Percona Platform will now be able to see their account Entitlements and Support tickets.
+    - **New Paid tier and special Advisor checks for customers**: PMM 2.27 introduces new sets of checks to cover additional database health assessments. All checks are grouped into sets of Advisors, according to the functionality and recommendations they provide. 
+
+* Advisors
+    - New presentation and tiers, including the Paid tier 
+    - Advisor checks have been extended with two new query types: getDiagnosticData and replSetGetStatus. 
+    - Documentation about developing checks 
+    - DBaaS - Simplified experience in registering EKS cluster with kubeconfig generated by eksctl.
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:39]: it is not and Advisor part

**Review by denisok — APPROVED** [review]

**Review by catalinaadam — CHANGES_REQUESTED** [review]
Implemented feedback from Roma

File: docs/release-notes/2.27.0.md (hunk 2 above)
Comment (denisok) [reply docs/release-notes/2.27.0.md:49]: it is gone :(

Comment (rnovikovP) [reply docs/release-notes/2.27.0.md:49]: Even if we mention features in Highlights,  we need to keep them in the general list with features/improvements/bugs.  So highlights are the top replacement but a summary of everything bellow

File: docs/release-notes/2.27.0.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+- Improved the documentation around [developing checks](https://github.com/percona/pmm-doc/blob/NewMongoDBChecks/docs/details/develop-checks.md).
+
+### Components upgrade
+  - Grafana: PMM 2.27.0 has now migrated to Grafana 8.3.5. This version of Grafana is loaded with a gamut of exciting features. For more information, see [What's new in Grafana v8.0](https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v8-0/).
+  - VictoriaMetrics: VictoriaMetrics has been upgraded to 1.72.0.
+
+
+## New Features
+
+- [PMM-9718](https://jira.percona.com/browse/PMM-9718): Federated connections to Percona Platform.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:51]: This is something like "changes  to PMM to support SocialLogin  for Percona account and ability to connect PMM to the portal."

Need to be at the bottom of the section. Probably moved to Improvements as the "connect to portal" feature already exists.

File: docs/release-notes/2.27.0.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Additional abilities for Mongo MongoDB Advisor Checks.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:53]:
```suggestion
- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Advisors: Additional abilities for Mongo MongoDB Advisor Checks.
```

File: docs/release-notes/2.27.0.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+- [PMM-9305](https://jira.percona.com/browse/PMM-9305), [PMM-8661](https://jira.percona.com/browse/PMM-8661): Visibility over Percona Platform Entitlements and Support Tickets.
+
+- [PMM-8434](https://jira.percona.com/browse/PMM-8434): Support for passing PMM Server Public Address as an environment variable while starting the PMM server.
+
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:58]: Will be good to add link to documentation with such parameters

File: docs/release-notes/2.27.0.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+- [PMM-9648](https://jira.percona.com/browse/PMM-9648): With PMM 2.27.0 migrating to Grafana 8.0, the Singlestat Panel has been deprecated and replaced with Stat Panel in Grafana for an enhanced user experience.
+
+- [PMM-9635](https://jira.percona.com/browse/PMM-9635): Broader database health checks with Advisors.
+
+- [PMM-9542](https://jira.percona.com/browse/PMM-9542): PMM now predefines certain flags that allow users to set all other [VictoriaMetrics parameters](https://docs.victoriametrics.com/#list-of-command-line-flags) as environment variables. For more information see [Documentation](https://www.percona.com/doc/percona-monitoring-and-management/2.x/details/victoria-metrics.html#environment-variables).
+
+- [PMM-9452](https://jira.percona.com/browse/PMM-9452): With this release of PMM, we have implemented a simplified password change method for the default admin user using the command line parameter change-admin-password.
+
+- [PMM-9339](https://jira.percona.com/browse/PMM-9339): The Use TLS option in webhook settings has been renamed to Show TLS setting to better reflect its functionality.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:70]:
```suggestion
- [PMM-9339](https://jira.percona.com/browse/PMM-9339): Integrated Alerting: The Use TLS option in webhook settings has been renamed to Show TLS setting to better reflect its functionality.
```

File: docs/release-notes/2.27.0.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+ 
+ - [PMM-8794](https://jira.percona.com/browse/PMM-8794): For consistency, we have implemented a unified 24 hours time format for backup management.
+ 
+- [PMM-8045](https://jira.percona.com/browse/PMM-8045): DBaaS: With this version of PMM, we have added a warning about the deletion of API keys so that the user is forewarned before deleting the API key.
+  
+- [PMM-4466](https://jira.percona.com/browse/PMM-4466): With this release of PMM we have migrated Grafana from using SQLite to PostgreSQL datasource for better performance of Auth sessions.
+
+## Bugs Fixed
+
+ - [PMM-9797](https://jira.percona.com/browse/PMM-9797): Fixed an issue where the data on the Home Dashboard was represented incorrectly on the Stat Panel, which could potentially confuse the user.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:88]:
```suggestion
 - [PMM-9797](https://jira.percona.com/browse/PMM-9797): Dashboards: Fixed an issue where the data on the Home Dashboard was represented incorrectly on the Stat Panel, which could potentially confuse the user.
```

File: docs/release-notes/2.27.0.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+ 
+ - [PMM-9783](https://jira.percona.com/browse/PMM-9783): Fixed an issue where QAN failed to work after an upgrade.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:90]:
```suggestion
 - [PMM-9783](https://jira.percona.com/browse/PMM-9783): Query Analytics: Fixed an issue where QAN failed to work after an upgrade.
```

File: docs/release-notes/2.27.0.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+ 
+ - [PMM-9757](https://jira.percona.com/browse/PMM-9757): Fixed an issue where the metrics for MongoDB were not being exposed when the database connection was getting lost.
+ 
+ - [PMM-9692](https://jira.percona.com/browse/PMM-9692): Ensured that Technical preview disclaimer message is always displayed in the Percona Platform section in the PMM Settings. 
+ 
+- [PMM-9671](https://jira.percona.com/browse/PMM-9671): Fixed an issue where the PMM configuration was lost due to pmm-managed crashing when the PMM agent version was null.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:96]: Here we need a real user explanation. 
Our users don't care/know about  internal components like pmm-amanged

File: docs/release-notes/2.27.0.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+ 
+- [PMM-9661](https://jira.percona.com/browse/PMM-9661): Fixed an issue where QAN layout breaks while resizing the window.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:98]:
```suggestion
- [PMM-9661](https://jira.percona.com/browse/PMM-9661): Query Analytics: Fixed an issue where QAN layout breaks while resizing the window.
```

File: docs/release-notes/2.27.0.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+ 
+- [PMM-9603](https://jira.percona.com/browse/PMM-9603): Fixed “Insufficient access rights” error that was displayed  for admin user after connecting  PMM server connect to Percona Portal. 
+  
+- [PMM-9570](https://jira.percona.com/browse/PMM-9570): DBaaS - Create DB Cluster dialog box was closing automatically while trying to create  a database cluster and had to be opened twice. This issue has been fixed now.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:102]:
```suggestion
- [PMM-9570](https://jira.percona.com/browse/PMM-9570): DBaaS: Create DB Cluster dialog box was closing automatically while trying to create  a database cluster and had to be opened twice. This issue has been fixed now.
```

File: docs/release-notes/2.27.0.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+ 
+- [PMM-9496](https://jira.percona.com/browse/PMM-9496): Fixed an issue where the host and database were not being monitored in DBaas.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:104]:
```suggestion
- [PMM-9496](https://jira.percona.com/browse/PMM-9496): DBaaS: Fixed an issue where the host and database were not being monitored in DBaas.
```

File: docs/release-notes/2.27.0.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+ 
+ - [PMM-9413](https://jira.percona.com/browse/PMM-9413): Fixed an issue where the PMM management daemon was getting deadlocked when a PMM agent was getting connected with a duplicate agent_id.
+ 
+ - [PMM-9323](https://jira.percona.com/browse/PMM-9323): Ensured PMM no longer downloads checks and templates from Percona Portal when the Telemetry option is disabled in the PMM Advanced Settings.
+ 
+ - [PMM-9015](https://jira.percona.com/browse/PMM-9015): Fixed an issue where PMM does not display when PostgreSQL is down, thus failing to capture the metrics for PostgreSQL.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:110]:
```suggestion
 - [PMM-9015](https://jira.percona.com/browse/PMM-9015): Dashboards: Fixed an issue where PMM does not display when PostgreSQL is down, thus failing to capture the metrics for PostgreSQL.
```

File: docs/release-notes/2.27.0.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+ 
+ - [PMM-8986](https://jira.percona.com/browse/PMM-8986): Fixed an issue where the Advisors check on the PMM servers monitoring a large number of database services was causing a timeout.
```
Comment (rnovikovP) [inline docs/release-notes/2.27.0.md:112]:
```suggestion
 - [PMM-8986](https://jira.percona.com/browse/PMM-8986): Advisors: Fixed an issue where the Advisors check on the PMM servers monitoring a large number of database services was causing a timeout. 
```

**Review by rnovikovP — CHANGES_REQUESTED** [review]
pls see  something like  https://www.percona.com/doc/percona-monitoring-and-management/2.x/release-notes/2.24.0.html

Added prefixes  to some issues and we also need to group them by prefixes in groups. 
 Ex:  

>  bugs : 
> advisor: ....
> advisor:
> dashoaards: .. 
>  everything else 

to keep better context for the reader

File: docs/release-notes/2.27.0.md (hunk 2 above)
Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md:49]: ok. But won't it be redundant?

File: docs/release-notes/2.27.0.md (hunk 4 above)
Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md]: Updated.

File: docs/release-notes/2.27.0.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Additional abilities for Mongo MongoDB Advisor Checks.
+
+- [PMM-9305](https://jira.percona.com/browse/PMM-9305), [PMM-8661](https://jira.percona.com/browse/PMM-8661): Visibility over Percona Platform Entitlements and Support Tickets.
+
+- [PMM-8434](https://jira.percona.com/browse/PMM-8434): Support for passing PMM Server Public Address as an environment variable while starting the PMM server.
+
+
+## Improvements
+
+- [PMM-9648](https://jira.percona.com/browse/PMM-9648): With PMM 2.27.0 migrating to Grafana 8.0, the Singlestat Panel has been deprecated and replaced with Stat Panel in Grafana for an enhanced user experience.
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md:66]: ok. Done

File: docs/release-notes/2.27.0.md (hunk 11 above)
Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md:111]: @puneet0191 Could you please provide inputs on this...

File: docs/release-notes/2.27.0.md (hunk 2 above)
Comment (denisok) [reply docs/release-notes/2.27.0.md:56]: highlights gives more info, list of the tickets only titles

File: docs/release-notes/2.27.0.md (hunk 7 above)
Comment (denisok) [reply docs/release-notes/2.27.0.md:65]: yep, added here: https://github.com/percona/pmm-doc/pull/730  @rasika-chivate could you link that do there?

File: docs/release-notes/2.27.0.md (hunk 11 above)
Comment (denisok) [reply docs/release-notes/2.27.0.md:111]: this issue could be seen by user as well, it affects PMM. From user perspective it is defined in a ticket description. Title could look something like: Update to 2.26.0 failed, PMM is crashing.

Comment (rasika-chivate) [reply docs/release-notes/2.27.0.md]: @denisok How about this?

Fixed an issue where an upgrade to 2.26.0 failed as PMM crashed.

**Review by taraskozub-percona — APPROVED** [review]

**Review by catalinaadam — APPROVED** [review]

File: docs/release-notes/index.md [hunk 18]
```diff
@@ -1,5 +1,6 @@
 # Release Notes
 
+- [Percona Monitoring and Management 2.27.0](2.27.0.md)
```
Comment (atymchuk) [inline docs/release-notes/index.md:3]: We need to fix this:
The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - release-notes/2.27.0.md
  - using/advisors/innodb_log_buffer_flushed_to_disk_after_each_trx.md
  
I do not see the 2.27.0.md in the navigation
Screenshot 2022-04-12 at 12 58 01

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: docs/release-notes/index.md (hunk 18 above)
Comment (rasika-chivate) [reply docs/release-notes/index.md:3]: @atymchuk I haven't made changes in the mkdocs-base.yml yet. I will be raising a separate PR for that.

File: docs/release-notes/2.27.0.md [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+
+  For more information, see [Working with Advisor checks](https://www.percona.com/doc/percona-monitoring-and-management/2.x/details/checks/index.html).
+
+
+- Advisor checks have been extended with two new query types: *GetDiagnosticData* and *replSetGetStatus*. 
+
+- Improved the documentation around [developing checks](h).
+
+
+### DBaas
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:71]:
```suggestion
### DBaaS
```

File: docs/release-notes/2.27.0.md [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+
+## New Features
+
+- [PMM-9718](https://jira.percona.com/browse/PMM-9718): PMM and Percona Portal Integration: Federated connections to Percona Platform.
+
+- [PMM-9305](https://jira.percona.com/browse/PMM-9305), [PMM-8661](https://jira.percona.com/browse/PMM-8661): PMM and Percona Portal Integration: Visibility over Percona Platform Entitlements and Support Tickets.
+
+- [PMM-9473](https://jira.percona.com/browse/PMM-9473): Advisors: Additional abilities for MongoDB Advisor Checks.
+
+- [PMM-8800](https://jira.percona.com/browse/PMM-8800): DBaas: With PMM, you can now have a simplified experience in registering your EKS cluster with kubeconfig generated by eksctl. Copy-paste the configuration by selecting Using Amazon Elastic Kubernetes Service (EKS) checkbox, and your K8s cluster is registered.
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:89]:
```suggestion
- [PMM-8800](https://jira.percona.com/browse/PMM-8800): DBaaS: With PMM, you can now have a simplified experience in registering your EKS cluster with kubeconfig generated by eksctl. Copy-paste the configuration by selecting Using Amazon Elastic Kubernetes Service (EKS) checkbox, and your K8s cluster is registered.
```

File: docs/release-notes/2.27.0.md [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+
+- [PMM-9603](https://jira.percona.com/browse/PMM-9603): PMM and Percona Portal Integration: Fixed “Insufficient access rights” error that was displayed  for admin user after connecting  PMM server connect to Percona Portal.
+  
+- [PMM-9323](https://jira.percona.com/browse/PMM-9323): PMM and Percona Portal Integration: Ensured PMM no longer downloads checks and templates from Percona Portal when the Telemetry option is disabled in the PMM Advanced Settings.
+
+- [PMM-8986](https://jira.percona.com/browse/PMM-8986): Advisors: Fixed an issue where the Advisors check on the PMM servers monitoring a large number of database services was causing a timeout. 
+
+- [PMM-9570](https://jira.percona.com/browse/PMM-9570): DBaaS: Create DB Cluster dialog box was closing automatically while trying to create  a database cluster and had to be opened twice. This issue has been fixed now.
+
+- [PMM-9496](https://jira.percona.com/browse/PMM-9496): DBaas: Fixed an issue where the host and database were not being monitored in DBaas.
```
Comment (denisok) [inline docs/release-notes/2.27.0.md:139]:
```suggestion
- [PMM-9496](https://jira.percona.com/browse/PMM-9496): DBaaS: Fixed an issue where the host and database were not being monitored in DBaas.
```

**Review by denisok — CHANGES_REQUESTED** [review]

**Review by denisok — APPROVED** [review]

**Review by catalinaadam — APPROVED** [review]

**Review by rnovikovP — APPROVED** [review]

**Pull request merged**